cmd/tendermint/commands/key_migrate.go

```diff
@@ -6,70 +6,71 @@ import (
 
 	"github.com/spf13/cobra"
 
-	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/config"
 	"github.com/tendermint/tendermint/libs/log"
 	"github.com/tendermint/tendermint/scripts/keymigrate"
 	"github.com/tendermint/tendermint/scripts/scmigrate"
 )
 
-func MakeKeyMigrateCommand(conf *cfg.Config, logger log.Logger) *cobra.Command {
+func MakeKeyMigrateCommand(conf *config.Config, logger log.Logger) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "key-migrate",
 		Short: "Run Database key migration",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			ctx, cancel := context.WithCancel(cmd.Context())
-			defer cancel()
+			return RunDatabaseMigration(cmd.Context(), logger, conf)
+		},
+	}
 
-			contexts := []string{
-				// this is ordered to put
-				// the more ephemeral tables first to
-				// forclose the possiblity of the
-				// ephemeral data overwriting later data
-				"tx_index",
-				"peerstore",
-				"light",
-				"blockstore",
-				"state",
-				"evidence",
-			}
+	// allow database info to be overridden via cli
+	addDBFlags(cmd, conf)
 
-			for idx, dbctx := range contexts {
-				logger.Info("beginning a key migration",
-					"dbctx", dbctx,
-					"num", idx+1,
-					"total", len(contexts),
-				)
+	return cmd
+}
 
-				db, err := cfg.DefaultDBProvider(&cfg.DBContext{
-					ID:     dbctx,
-					Config: conf,
-				})
+func RunDatabaseMigration(ctx context.Context, logger log.Logger, conf *config.Config) error {
```

Comment:
I'd prefer if the function signature was just `db-dir` and `db-backend` instead of the entire config

Reply:
I don’t think we can do this because the db provider creator asks for the entire config object.

Reply:
Ok that's a bit annoying if it just uses those two values but it's not too important. As context, I want to help billy with gaia to use this command in their binary

Reply:
yeah, changing that is a bit of churn as the db context is used in a lot of place. and I can look at doing that, but probably better to do it in a different PR.